CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -92,6 +92,7 @@ @implementation CleverPush
 NSMutableArray* pendingChannelConfigListeners;
 NSMutableArray* pendingSubscriptionListeners;
 NSMutableArray* pendingTrackingConsentListeners;
+NSMutableArray* subscriptionTags;
```

Comment:
Why do we have a global variable here? Please remove it

Reply:
because to keep the reference of the tags inside the blocks, I've define global variable here.
and if we define variable inside the blocks we will have to define something like this
`        __block NSMutableArray* subscriptionTags = [NSMutableArray arrayWithArray:[userDefaults arrayForKey:@"CleverPush_SUBSCRIPTION_TAGS"]];
`
and that `__block NSMutableArray` and block not retain the values of the tags which we have added during the api calls. It is always allocates its memory reference and keep the only one object except the bunch of data.

Reply:
I am not happy with this solution.

It still might happen that a user calls "addSubscriptionTag" many times at once. I guess a global variable will cause issues and race conditions here.

Reply:
Please find a way to keep this in the method scope